src/core/painting/qgsgeometrypaintdevice.cpp

```diff
@@ -421,20 +481,28 @@ void addSubpathGeometries( QgsGeometryCollection &collection, const QPainterPath
   if ( currentX.size() > 1 )
   {
     std::unique_ptr< QgsLineString > line = std::make_unique< QgsLineString >( currentX, currentY );
-    if ( line->isClosed() )
+    if ( mUsePathStroker )
     {
+      addStrokedLine( line.get(), penWidth, endCapStyle, joinStyle, miterLimit, transformIsIdentity ? nullptr : &matrix );
+    }
+    else if ( line->isClosed() )
+    {
+      if ( !transformIsIdentity )
+        line->transform( matrix );
       queuedPolygons.emplace_back( std::make_unique< QgsPolygon >( line.release() ) );
     }
     else
     {
-      collection.addGeometry( line.release() );
+      if ( !transformIsIdentity )
+        line->transform( matrix );
+      mGeometry.addGeometry( line.release() );
     }
   }
 
   if ( queuedPolygons.empty() )
     return;
 
-  collection.reserve( collection.numGeometries() + queuedPolygons.size() );
+  mGeometry.reserve( mGeometry.numGeometries() + queuedPolygons.size() );
```

Comment:
```suggestion
  mGeometry.reserve( mGeometry.numGeometries() + static_cast<int>( queuedPolygons.size() ) );
```
To fix the clang-tidy issue